chapters/arrays.tex

```diff
@@ -1268,10 +1268,20 @@ \subsection{Element-wise Division}\label{array-element-wise-division}\label{elem
 
 \subsection{Element-wise Exponentiation}\label{element-wise-exponentiation}
 
-Exponentiation \lstinline!a ^ b! is defined as \lstinline[language=C]!pow(double a, double b)! in the ANSI~C library if both \lstinline!a! and \lstinline!b! are
-\lstinline!Real! scalars. A \lstinline!Real! scalar value is returned.  If \lstinline!a! or \lstinline!b! are \lstinline!Integer! scalars, they are
-automatically promoted to \lstinline!Real!.  Consequences of exceptional situations, such as ($\text{\lstinline!a!} = 0.0$ and $\text{\lstinline!b!} \leq 0.0$,
-$\text{\lstinline!a!} < 0$ and \lstinline!b! is not an integer) or overflow are undefined.
+Exponentiation \lstinline!a ^ b! always returns a \lstinline!Real! scalar value, and it is required that \lstinline!a! and \lstinline!b! are scalar \lstinline!Real! or \lstinline!Integer! values.
+The result should correspond to mathematical exponentiation (ideally the correctly rounded result based on the inputs) with the following special cases:
+\begin{itemize}
+\item If $\text{\lstinline!a!} = 0.0$ and $\text{\lstinline!b!} = 0$ for an \lstinline!Integer! expression \lstinline!b! the result is $1.0$.
+\item If $\text{\lstinline!a!} < 0$ and \lstinline!b! is an \lstinline!Integer!, the result is defined as $\pm |a|^b$, with sign depending on whether \lstinline!b! is even (positive) or odd (negative).
+\item If $\text{\lstinline!a!} < 0$ and \lstinline!b! is a \lstinline!Real! having a non-zero integer value, the result is defined as $\pm |a|^b$, with sign depending on whether \lstinline!b! is even (positive) or odd (negative). This case is deprecated, and tools may warn once during a simulation if this occurs.
```

Comment:
Line break:
```suggestion
\item
  If $\text{\lstinline!a!} < 0$ and \lstinline!b! is a \lstinline!Real! having a non-zero integer value, the result is defined as $\pm |a|^b$, with sign depending on whether \lstinline!b! is even (positive) or odd (negative).
  This case is deprecated, and tools may warn once during a simulation if this occurs.
```

Reply:
Apart from the line break, what is the right amount of deprecation detail?  Wouldn't it be better to just say it's deprecated?
```suggestion
\item A deprecated semantics is to treat $\text{\lstinline!a!} < 0$ and a \lstinline!Real! \lstinline!b! having a non-zero integer value as if \lstinline!b! were an \lstinline!Integer!.
```

Reply:
Ok, avoids duplication and deprecated semantics referencing non-deprecated ones should be ok.